File: VB/E1554.Module.Win/WinModule.vb
Imports System
Imports System.ComponentModel

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.ExpressApp.Win.SystemModule

Namespace E1554.Module.Win
	<ToolboxItemFilter("Xaf.Platform.Win")>
	Public NotInheritable Partial Class E1554WindowsFormsModule
		Inherits ModuleBase

		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub Application_CreateCustomModelDifferenceStore(ByVal sender As Object, ByVal e As CreateCustomModelDifferenceStoreEventArgs)
			e.Store = New ModelDifferenceDbStore(DirectCast(sender, XafApplication), GetType(ModelDifference), True)
			e.Handled = True
		End Sub
		Private Sub Application_CreateCustomUserModelDifferenceStore(ByVal sender As Object, ByVal e As CreateCustomModelDifferenceStoreEventArgs)
			e.Store = New ModelDifferenceDbStore(DirectCast(sender, XafApplication), GetType(ModelDifference), False)
			e.Handled = True
		End Sub
		Public Overrides Sub Setup(ByVal application As XafApplication)
			MyBase.Setup(application)
			AddHandler application.ModelChanged, AddressOf application_ModelChanged
			AddHandler application.CreateCustomModelDifferenceStore, AddressOf Application_CreateCustomModelDifferenceStore
			AddHandler application.CreateCustomUserModelDifferenceStore, AddressOf Application_CreateCustomUserModelDifferenceStore
		End Sub

		Private Sub application_ModelChanged(ByVal sender As Object, ByVal e As EventArgs)
			Dim uiType As UIType = DirectCast(Application.Model.Options, IModelOptionsWin).UIType
			If uiType = UIType.StandardMDI Then
				Application.ShowViewStrategy = New MyMdiShowViewStrategy(Application, DevExpress.ExpressApp.Win.Templates.MdiMode.Standard)
			ElseIf uiType = UIType.TabbedMDI Then
				Application.ShowViewStrategy = New MyMdiShowViewStrategy(Application, DevExpress.ExpressApp.Win.Templates.MdiMode.Tabbed)
			End If
		End Sub
	End Class
End Namespace

File: CS/E1554.Module.Win/WinModule.cs
using System;
using System.ComponentModel;

using DevExpress.ExpressApp;
using DevExpress.Persistent.BaseImpl;
using DevExpress.ExpressApp.Win.SystemModule;

namespace E1554.Module.Win {
    [ToolboxItemFilter("Xaf.Platform.Win")]
    public sealed partial class E1554WindowsFormsModule : ModuleBase {
        public E1554WindowsFormsModule() {
            InitializeComponent();
        }
        private void Application_CreateCustomModelDifferenceStore(Object sender, CreateCustomModelDifferenceStoreEventArgs e) {
            e.Store = new ModelDifferenceDbStore((XafApplication)sender, typeof(ModelDifference), true);
            e.Handled = true;
        }
        private void Application_CreateCustomUserModelDifferenceStore(Object sender, CreateCustomModelDifferenceStoreEventArgs e) {
            e.Store = new ModelDifferenceDbStore((XafApplication)sender, typeof(ModelDifference), false);
            e.Handled = true;
        }
        public override void Setup(XafApplication application) {
            base.Setup(application);
            application.ModelChanged += new EventHandler(application_ModelChanged);
            application.CreateCustomModelDifferenceStore += Application_CreateCustomModelDifferenceStore;
            application.CreateCustomUserModelDifferenceStore += Application_CreateCustomUserModelDifferenceStore;
        }

        void application_ModelChanged(object sender, EventArgs e) {
            UIType uiType = ((IModelOptionsWin)Application.Model.Options).UIType;
            if (uiType == UIType.StandardMDI) {
                Application.ShowViewStrategy = new MyMdiShowViewStrategy(Application, DevExpress.ExpressApp.Win.Templates.MdiMode.Standard);
            } else if (uiType == UIType.TabbedMDI) {
                Application.ShowViewStrategy = new MyMdiShowViewStrategy(Application, DevExpress.ExpressApp.Win.Templates.MdiMode.Tabbed);
            }
        }
    }
}
